cpp/src/arrow/telemetry/telemetry_test.cc

```diff
@@ -78,6 +81,10 @@ void Log(LogLevel severity, std::string_view message) {
 class TestLogging : public ::testing::Test {
  public:
   void SetUp() override {
+    auto env_var = arrow::internal::GetEnvVar(kLoggingEnvVar);
+    if (env_var.status().IsKeyError()) {
+      GTEST_SKIP() << "Env var " << kLoggingEnvVar << " is undefined";
+    }
```

Comment:
If these tests need an env var to be set, can it be set 1) either in the test setup (if not too late) 2) or in `ci/scripts/cpp_test.sh`? It would be better than skipping this test on many different CI configurations.

(it would probably be better if the tests could be made independent of the environment, btw, but I realize that may be difficult depending on implementation specifics)